src/azul/security/verifier.py

```diff
@@ -0,0 +1,112 @@
+# The code is based on dss.util.security.
+from base64 import urlsafe_b64decode
+import logging
+from urllib.parse import urlparse
+
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.backends import default_backend
+import jwt
+from jwt.exceptions import DecodeError, PyJWTError
+import requests
+
+from azul import config
+
+logger = logging.getLogger(__name__)
+
+allowed_algorithms = ('RS256',)
+gserviceaccount_domain = "iam.gserviceaccount.com"
+
+# Shared HTTP session
+def shared_session():
```

Comment:
The name of the method and the (redundant) comment are misleading. There is no sharing going on here. Every invocation of a function in this module will create a new session.

Reply:
This is moved to `azul.security.authenticator.Authenticator.session`.